Web.md

```diff
@@ -22,6 +22,82 @@ and retrieve compiled modules from offline storage, instantiate compiled modules
 with JavaScript imports, call the exported functions of instantiated modules,
 alias the exported memory of instantiated modules, etc.
 
+In addition to the core JavaScript API, the Web embedding includes additional
+methods useful in that context.
+Non-web embeddings are not required to support these additional methods.
+
+### Additional Web Embedding API
+
+#### `WebAssembly.compileResponse`
+
+The `compileResponse` function is overloaded based on types of its arguments.
+If none of the following overloads match, then the returned `Promise` is
+[rejected](http://tc39.github.io/ecma262/#sec-rejectpromise)
+with a [`TypeError`](https://tc39.github.io/ecma262/#sec-native-error-types-used-in-this-standard-typeerror).
+
+```
+Promise<WebAssembly.Module> compileResponse(Response source)
+
+Promise<WebAssembly.Module> compileResponse(Promise<Response> source)
+```
+
+The source of the material to compile can be
+a [`Response`](https://fetch.spec.whatwg.org/#response-class)
+or a Promise of a
+[`Response`](https://fetch.spec.whatwg.org/#response-class).
+
+This function starts an asynchronous task to compile a `WebAssembly.Module`
+as described in the [`WebAssembly.Module` constructor](#webassemblymodule-constructor).
+On success, the `Promise` is [fulfilled](http://tc39.github.io/ecma262/#sec-fulfillpromise)
+with the resulting `WebAssembly.Module` object. On failure, the `Promise` is
+[rejected](http://tc39.github.io/ecma262/#sec-rejectpromise) with a
+`WebAssembly.CompileError`.
+
+The `Response` or `Promise<Response>` is used as the source of the
+bytes to compile.
+WebAssembly `source` data is expected to have a mime type of `application/wasm`.
+Other mime types
+[reject](http://tc39.github.io/ecma262/#sec-rejectpromise) the Promise with a
+`WebAssembly.CompileError`.
+
+#### `WebAssembly.instantiateResponse`
+
+The `instantiate` function is overloaded based on types of its arguments.
+If none of the following overloads match, then the returned `Promise` is
+[rejected](http://tc39.github.io/ecma262/#sec-rejectpromise)
+with a [`TypeError`](https://tc39.github.io/ecma262/#sec-native-error-types-used-in-this-standard-typeerror).
+
+```
+Promise<{module:WebAssembly.Module, instance:WebAssembly.Instance}>
+  instantiateResponse(Response source [, importObject])
+
+Promise<{module:WebAssembly.Module, instance:WebAssembly.Instance}>
+  instantiateResponse(Promise<Response> source [, importObject])
+```
```

Comment:
Agreed.

> If we did it for the one-step instantiate methods, it would probably end up requiring a declarative way to describe all the hookups (which seems to re-do a lot of the ideas from the ES6 modules space).

IIUC, there is a JS function to dynamically load an ES6 module proposed now (I forget the name).  With wasm modules integrated with ES6 modules, that would basically give us this function.